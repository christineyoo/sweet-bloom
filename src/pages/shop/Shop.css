.shop-flex {
  display: flex;
  flex-direction: column;
}

.shop-flex-1 {
  flex: 1;
  padding: 0.5rem;
  text-align: center;
}

#shop {
  padding: 2rem;
}

#shop > h1 {
  font-weight: 900;
  text-align: center;
  color: #83364b;
}

#shop-button {
  padding: 1rem 4rem;
  background: transparent;
  border: 1px solid #83364b;
  font-family: 'Poppins', sans-serif;
  color: #83364b;
  font-weight: 900;
  transition: background 0.5s ease-in-out;
  margin: 0.5rem;
}

#shop-button:hover {
  background-color: #83364b;
  color: #ffffff;
  cursor: pointer;
}

@media all and (min-width: 800px) {
  #shop {
    padding: 5% 10%;
  }

  .shop-flex {
    flex-direction: row;
    flex-wrap: wrap;
  }

  .shop-flex-1 {
    width: 100%;
    height: 100%;
  }
}
