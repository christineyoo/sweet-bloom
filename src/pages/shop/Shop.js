/* eslint-disable no-unreachable */
import React, { Component } from 'react';
import ApiContext from '../../ApiContext';
import { Link } from 'react-router-dom';
import ItemLink from '../../organisms/itemLink/ItemLink';
import './Shop.css';

class Shop extends Component {
  static contextType = ApiContext;

  scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth'
    });
  };

  renderGroups = () => {
    const groupsArray = [
      { id: 1, group_name: 'Flowers' },
      { id: 2, group_name: 'Desserts' }
    ];
    return (
      <div style={{ textAlign: 'center' }}>
        <Link to={`/group/${groupsArray[0].id}`}>
          <button id='shop-button'>{groupsArray[0].group_name}</button>
        </Link>
        <Link to={`/group/${groupsArray[1].id}`}>
          <button id='shop-button'>{groupsArray[1].group_name}</button>
        </Link>
      </div>
    );
  };

  renderItems = () => {
    const copyItems = this.context.items || [];
    return (
      <div className='shop-flex'>
        {copyItems.map((item, i) => {
          return (
            <ItemLink
              key={i}
              item_id={item.id}
              item_name={item.item_name}
              item_price={item.item_price}
              item_url={item.item_url}
              item_description={item.item_description}
            />
          );
        })}
      </div>
    );
  };

  render() {
    return (
      <div id='shop'>
        <h1>Shop All Items</h1>
        {this.renderGroups()}
        {this.renderItems()}
        <button id='shop-button' onClick={() => this.scrollToTop()}>Back to top</button>
      </div>
    );
  }
}

export default Shop;
