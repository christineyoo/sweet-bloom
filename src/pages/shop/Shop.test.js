import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Shop from './Shop';

it('renders Shop UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Shop />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});