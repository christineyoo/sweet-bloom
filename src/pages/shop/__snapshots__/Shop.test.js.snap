// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Shop UI as expected 1`] = `
<div
  id="shop"
>
  <h1>
    All Items
  </h1>
  <h2>
    Browse through our wide collection of floral bouquets and desserts.
  </h2>
  <h3>
    Filter by 
    <div>
      <a
        href="/group/1"
        onClick={[Function]}
      >
        <button
          id="shop-button"
        >
          Flowers
        </button>
      </a>
      <a
        href="/group/2"
        onClick={[Function]}
      >
        <button
          id="shop-button"
        >
          Desserts
        </button>
      </a>
    </div>
  </h3>
  <div
    className="shop-flex"
  />
  <button
    onClick={[Function]}
  >
    Back to top
  </button>
</div>
`;
